src/main/xerus/monstercat/api/Player.kt

```diff
@@ -90,7 +89,16 @@ object Player: FadingHBox(true, targetHeight = 25) {
 			addButton { reset() }.id("back")
 			fill(pos = 0)
 			fill()
-			add(closeButton)
+			if (Playlist.tracks.size < 2){
+				add(closeButton)
+			}else{
+				add(buttonWithId("skip") { playNext() }).apply {
+					tooltip = Tooltip("Skip")
+				}
+				Timer("SkipErroredSong", false).schedule(5000) { // fixme : hardcoded delay in ms
```

Comment:
how does this work? I see no error check.

Reply:
If a song is not found, or not playable (`showBack` function), and we are in a playlist, we show a button to skip it, and will automatically skip it after 5 seconds to not interrupt the playlist.

Reply:
But how does this Timer respond to whether there was an error or not? To me it looks like it would just skip any song after 5 seconds.

Reply:
Timer is created if there is an error. Thus skips the errored song 5 seconds later. I should refactor this with a new function (something along the lines of onError ?)

Reply:
Function ``showBack`` is called if there is an error while playing a track

Reply:
Renamed ``showBack`` to a more explicit ``showError``

Reply:
See 2f5a7e8ce3bf9ac0aae50906cb6a7614a436028c